### PR TITLE
Fully abstracted file system, plus code quality improvements.

### PR DESCRIPTION
**Pull request by MasterEric**
This PR performs the following changes:

* Replaced static access to `PolymodFileSystem` with access to an instance of IFileSystem. This defaults to SysFileSystem if available.
* Added a parameter to allow developers to pass a custom IFileSystem, similar to IBackend.
  * Developers should now be able to implement #12 on their own, although built-in implementations of stuff like zip modloading would be cool.
* Added a VSCode config to tell users to install the Haxe extensions for code formatting and syntax highlighting.
* Added a HaxeFormatter config and applied it to all files.

### CODE REVIEW

**Comment by MasterEric** [comment]
Hey Lars,

I wanted to say thank you for all the work you've put into this project (especially recently, [I have a basic implementation of the latest changes working in Friday Night Funkin' already](https://www.youtube.com/watch?v=1A5ChNEf5F4)). It's very powerful and flexible as is, and every bit of effort you put into making it better is much appreciated.

In terms of the code formatting changes, I put them in their own commit, but I realize now that the "Files changed" view doesn't let you filter by that. I'll move them to their own PR.

As for the Electron PR, I can test merging the branches together and see how that plays out.

**Comment by larsiusprime** [comment]
Thanks for working with me here :) I also just realized that some of my samples are out of date, such as the Heaps demo. Lot of maintenance work to catch up on, and I don't even have any unit tests, I just rely on the samples telling me if something seems broken or not.

File: polymod/Polymod.hx [hunk 1]
```diff
@@ -352,56 +377,6 @@ class Polymod
         }
     }
 
-    private static function getFileSystem()
```
Comment (MasterEric) [inline polymod/Polymod.hx:355]: This function is now obsolete. File system access is no longer static.

File: polymod/Polymod.hx [hunk 2]
```diff
@@ -352,56 +377,6 @@ class Polymod
...
-    {
-        #if sys
-        return new polymod.fs.SysFileSystem();
-        #end
-        return null;
-    }
-
-    private static function getMetadata(dir:String):ModMetadata
```
Comment (MasterEric) [inline polymod/Polymod.hx:363]: This function has been moved into the non-static IFileSystem object.

File: polymod/fs/IFileSystem.hx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * THE SOFTWARE.
+ * 
+ */
+
+package polymod.fs;
+
+import polymod.Polymod.ModMetadata;
+import haxe.io.Bytes;
+
+interface IFileSystem
```
Comment (MasterEric) [inline polymod/fs/IFileSystem.hx:29]: This class provides a new interface, which developers can implement on a class they pass as a parameter to Polymod. This allows them to override the base file system access class in a standard manner.

**Comment by MasterEric** [comment]
Hey, I've force pushed a new commit which removes the formatting changes.

As for the Electron support, it appears that the two changes are absolutely not compatible due to major merge conflicts. @TamarCurry adds a new NodeFileSystem but doesn't implement the interface I created here.

For this to work, we need to have one developer's changes be merged and have the other rework their PR to be compatible.

**Comment by larsiusprime** [comment]
Yeah, I figured as much. So I promised Tamar that I would look into his first as he's been waiting patiently for over a week now. I may need to get directly involved to deal with the cleanup work to reconcile his with yours, if you're okay with that.

**Comment by MasterEric** [comment]
That's fine with me, as long as it doesn't put too much work on your plate. 😄

**Comment by larsiusprime** [comment]
Eh, all in the day of a life of an open source maintainer. If you'll be patient with me we'll get this sorted eventually.

**Pull request merged**